From dad35dbd1456e917ecb57f4f8baed2c6a7fc17d1 Mon Sep 17 00:00:00 2001
From: Micha Ober <git@ober-mail.de>
Date: Mon, 29 Jan 2024 19:02:09 +0100
Subject: [PATCH] Disable x2/x4 read ops

---
 drivers/mtd/nand/spi/micron.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/nand/spi/micron.c b/drivers/mtd/nand/spi/micron.c
index f1be05f45..9ae636bcc 100644
--- a/drivers/mtd/nand/spi/micron.c
+++ b/drivers/mtd/nand/spi/micron.c
@@ -46,8 +46,8 @@ static SPINAND_OP_VARIANTS(x4_update_cache_variants,
 
 /* Micron  MT29F2G01AAAED Device */
 static SPINAND_OP_VARIANTS(x4_read_cache_variants,
-			   SPINAND_PAGE_READ_FROM_CACHE_X4_OP(0, 1, NULL, 0),
-			   SPINAND_PAGE_READ_FROM_CACHE_X2_OP(0, 1, NULL, 0),
+			   //SPINAND_PAGE_READ_FROM_CACHE_X4_OP(0, 1, NULL, 0),
+			   //SPINAND_PAGE_READ_FROM_CACHE_X2_OP(0, 1, NULL, 0),
 			   SPINAND_PAGE_READ_FROM_CACHE_OP(true, 0, 1, NULL, 0),
 			   SPINAND_PAGE_READ_FROM_CACHE_OP(false, 0, 1, NULL, 0));
 
-- 
2.34.1

